contracts/RelaySugar.vy

```diff
@@ -54,61 +64,154 @@ interface IAutoCompounder:
+
+    for venft_index in range(0, len(account_venfts)):
+      if managed_id == account_venfts[venft_index][0]:
+        account_venft_ids.append(account_venfts[venft_index][1])
 
     compounders.append(Relay({
+      venft_id: managed_id,
+      decimals: managed_nft.decimals,
+      amount: managed_nft.amount,
+      voting_amount: managed_nft.voting_amount,
+      voted_at: managed_nft.voted_at,
+      votes: managed_nft.votes,
+      token: managed_nft.token,
+      manager: manager,
+      compounder: addresses[index],
+      inactive: inactive,
       name: autocompounder.name(),
-      managed_nft: managed_nft,
-      address: addresses[index],
-      inactive: inactive
+      account_venft_ids: account_venft_ids
     }))
 
   return compounders
 
-@external
+@internal
 @view
-def deposits(_account: address) -> DynArray[Deposit, MAX_RESULTS]:
+def _deposits(_account: address) -> DynArray[VeNFT, MAX_RESULTS]:
   """
-  @notice Returns all of an account's Relay Deposits
+  @notice Returns all of an account's Relay veNFT deposits
   @param _account The account address
-  @return Array of Deposits
+  @return Array of VeNFTs
   """
-  deposits: DynArray[Deposit, MAX_RESULTS] = empty(DynArray[Deposit, MAX_RESULTS])
-  nfts: DynArray[VeNFT, MAX_RESULTS] = self.ve_sugar.byAccount(_account)
+  deposits: DynArray[VeNFT, MAX_RESULTS] = empty(DynArray[VeNFT, MAX_RESULTS])
+
+  if _account == empty(address):
+    return deposits
+
+  for index in range(MAX_RESULTS):
+    venft_id: uint256 = self.ve.ownerToNFTokenIdList(_account, index)
 
-  for index in range(0, len(nfts)):
-    manager_id: uint256 = self.ve.idToManaged(nfts[index].id)
+    if venft_id == 0:
+      break
+
+    manager_id: uint256 = self.ve.idToManaged(venft_id)
 
     if manager_id != 0:
-      deposits.append(Deposit({
-        deposited_nft: nfts[index],
-        manager_id: manager_id
-      }))
+      deposits.append(self._byId(venft_id))
 
   return deposits
+
+@internal
+@view
+def _byId(_id: uint256) -> VeNFT:
+  """
+  @notice Returns veNFT data based on ID
+  @param _id The index/ID to lookup
+  @return VeNFT struct
+  """
+  votes: DynArray[LpVotes, MAX_PAIRS] = []
+  amount: uint128 = 0
+  amount = self.ve.locked(_id)[0]
+  last_voted: uint256 = 0
+  manager_id: uint256 = self.ve.idToManaged(_id)
+
+  if self.ve.voted(_id):
+    last_voted = self.voter.lastVoted(_id)
+
+  vote_weight: uint256 = self.voter.usedWeights(_id)
+  # Since we don't have a way to see how many pools the veNFT voted...
+  left_weight: uint256 = vote_weight
+
+  for index in range(MAX_PAIRS):
+    if left_weight == 0:
+      break
+
+    lp: address = self.voter.poolVote(_id, index)
+
+    if lp == empty(address):
+      break
+
+    weight: uint256 = self.voter.votes(_id, lp)
+
+    votes.append(LpVotes({
+      lp: lp,
+      weight: weight
+    }))
+
+    # Remove _counted_ weight to see if there are other pool votes left...
+    left_weight -= weight
+
+  return VeNFT({
```

Comment:
I suggest we return here a `Relay` struct instead of the `VeNFT` one, since most of these attributes are the ones we need for a `Relay` instance.

Should simplify the code as well in the `_autocompounders()`